syntaxhighlighter.php

```diff
@@ -659,16 +659,21 @@ function shortcode_hack( $content, $callback, $ignore_html = true ) {
 			return $content;
 		}
 
-		// Backup current registered shortcodes and clear them all out
-		$orig_shortcode_tags = $shortcode_tags;
+		// Backup current registered shortcodes and clear them all out (we do not backup our own, because we will add and parse them below)
+		$orig_shortcode_tags = array_diff_key( $shortcode_tags, array_flip( $this->shortcodes ) );
 		remove_all_shortcodes();
 
 		// Register all of this plugin's shortcodes
 		foreach ( $this->shortcodes as $shortcode ) {
 			add_shortcode( $shortcode, $callback );
 		}
 
-		$regex = '/' . get_shortcode_regex( $this->shortcodes ) . '/';
+		// Register all other shortcodes, ensuring their content remains unchanged using yet another hack.
+		foreach ( $orig_shortcode_tags as $shortcode_tagname => $shortcode ) {
+			add_shortcode( $shortcode_tagname, array( $this, 'return_entire_shortcode_callback' ) );
+		}
+
+		$regex = '/' . get_shortcode_regex( array_merge( $this->shortcodes, array_keys( $orig_shortcode_tags ) ) ) . '/';
```

Comment:
You are 100% right. I wasn't aware that by default `get_shortcode_regex` returned all registered shortcodes. Since we are adding them all, the result is the same. We should remove the parameter.

Reply:
Removed in 736f10a.